### PR TITLE
More xml comments

### PR DESCRIPTION
**Pull request by Shane32**
Some unused methods have been marked obsolete.

### CODE REVIEW

File: src/GraphQL/DocumentWriterExtensions.cs [hunk 1]
```diff
@@ -4,6 +4,9 @@
 
 namespace GraphQL
 {
+    /// <summary>
+    /// Provides extension functions for document writers.
```
Comment (sungam3r) [inline src/GraphQL/DocumentWriterExtensions.cs:8]:
```suggestion
    /// Provides extension methods for document writers.
```

File: src/GraphQL/Execution/ExecutionNode.cs [hunk 2]
```diff
@@ -289,6 +289,10 @@ IEnumerable<ExecutionNode> IParentExecutionNode.GetChildNodes()
     /// </summary>
     public class RootExecutionNode : ObjectExecutionNode
     {
+        /// <summary>
+        /// Initializes a new instance for the specified root graph type.
+        /// </summary>
+        /// <param name="graphType"></param>
```
Comment (sungam3r) [inline src/GraphQL/Execution/ExecutionNode.cs:295]:
```suggestion
```

File: src/GraphQL/Execution/IExecutionContext.cs [hunk 3]
```diff
@@ -7,6 +7,9 @@
 
 namespace GraphQL.Execution
 {
+    /// <summary>
+    /// Provides information regarding the currently executing document.
```
Comment (sungam3r) [inline src/GraphQL/Execution/IExecutionContext.cs:11]: executed?

Comment (Shane32) [reply src/GraphQL/Execution/IExecutionContext.cs:11]: the context is (typically) only used while the document is being executed, not afterwards.

Comment (sungam3r) [reply src/GraphQL/Execution/IExecutionContext.cs:11]: right

**Review by sungam3r — APPROVED** [review]

**Pull request merged**